modular_zubbers/modules/blooper/atoms_movable.dm

```diff
@@ -0,0 +1,84 @@
+	var/blooper_id
+	var/blooper_pitch = 1
+	var/blooper_pitch_range = 0.2 //Actual pitch is (pitch - (blooper_pitch_range*0.5)) to (pitch + (blooper_pitch_range*0.5))
+	var/blooper_volume = 50
+	var/blooper_speed = 4 //Lower values are faster, higher values are slower
+	var/blooper_current_blooper //When bloopers are queued, this gets passed to the blooper proc. If blooper_current_blooper doesn't match the args passed to the blooper proc (if passed at all), then the blooper simply doesn't play. Basic curtailing of spam~
+
+/atom/movable/proc/set_blooper(id)
+	if(!id)
+		return FALSE
+	var/datum/blooper/B = GLOB.blooper_list[id]
+	if(!B)
+		return FALSE
+	blooper = sound(initial(B.soundpath))
+	blooper_id = id
+	return blooper
+
+/atom/movable/proc/blooper(list/listeners, distance, volume, pitch, queue_time)
+	if(!GLOB.blooper_allowed)
+		return
+	if(queue_time && blooper_current_blooper != queue_time)
+		return
+	if(!blooper)
+		if(!blooper_id || !set_blooper(blooper_id)) //just-in-time blooper generation
+			return
+	volume = min(volume, 100)
+	var/turf/T = get_turf(src)
+	for(var/mob/M in listeners)
+		M.playsound_local(T, vol = volume, vary = TRUE, frequency = pitch, max_distance = distance, falloff_distance = 0, falloff_exponent = BLOOPER_SOUND_FALLOFF_EXPONENT, sound_to_use = blooper, distance_multiplier = 1)
+
+/atom/movable/send_speech(message, range = 7, obj/source = src, bubble_type, list/spans, datum/language/message_language, list/message_mods = list(), forced = FALSE, tts_message, list/tts_filter)
+	. = ..()
+	var/list/listeners = get_hearers_in_view(range, source)
+	if(blooper || blooper_id)
+		for(var/mob/M in listeners)
+			if(!M.client)
+				continue
+			if(!(M.client?.prefs.read_preference(/datum/preference/toggle/hear_sound_blooper)))
+				listeners -= M
+		var/bloopers = min(round((LAZYLEN(message) / blooper_speed)) + 1, BLOOPER_MAX_BLOOPERS)
+		var/total_delay
+		blooper_current_blooper = world.time //this is juuuuust random enough to reliably be unique every time send_speech() is called, in most scenarios
+		for(var/i in 1 to bloopers)
+			if(total_delay > BLOOPER_MAX_TIME)
+				break
+			addtimer(CALLBACK(src, PROC_REF(blooper), listeners, range, blooper_volume, BLOOPER_DO_VARY(blooper_pitch, blooper_pitch_range), blooper_current_blooper), total_delay)
+			total_delay += rand(DS2TICKS(blooper_speed / BLOOPER_SPEED_BASELINE), DS2TICKS(blooper_speed / BLOOPER_SPEED_BASELINE) + DS2TICKS(blooper_speed / BLOOPER_SPEED_BASELINE)) TICKS
+
+/mob/living/send_speech(message_raw, message_range = 6, obj/source = src, bubble_type = bubble_icon, list/spans, datum/language/message_language = null, list/message_mods = list(), forced = null, tts_message, list/tts_filter)
+	. = ..()
+	if(client)
+		if(!(client.prefs.read_preference(/datum/preference/toggle/send_sound_blooper)))
+			return
+	blooper_volume = client.prefs.read_preference(/datum/preference/numeric/sound_blooper_volume) //volume scales with your volume slider in game preferences.
```

Comment:
Not what I suggested

```suggestion
	if(!client)
		return
	if(!(client.prefs.read_preference(/datum/preference/toggle/send_sound_blooper)))
		return
	blooper_volume = client.prefs.read_preference(/datum/preference/numeric/sound_blooper_volume) //volume scales with your volume slider in game preferences.
```

Reply:
This volume code sets the blooper volume for all listeners to whatever the sender has configured as volume.